04.ls/ls.rb

```diff
@@ -19,27 +20,18 @@
 
 def main
   options = { a: false, r: false, l: false }
-  path = './'
-  ARGV.each do |argv|
-    if FileTest.directory?(argv)
-      path = argv
-    elsif argv.start_with?('-')
-      specified_options = argv.delete('-').chars
-      specified_options.each do |specified_option|
-        case specified_option
-        when 'a' then options[:a] = true
-        when 'r' then options[:r] = true
-        when 'l' then options[:l] = true
-        end
-      end
-    end
-  end
+  opt = OptionParser.new
+  opt.on('-a') { options[:l] = true }
+  opt.on('-r') { options[:l] = true }
+  opt.on('-l') { options[:l] = true }
+  argv = opt.parse(ARGV)
```

Comment:
pathの配列が渡されることを前提にしているなら

```suggestion
  paths = opt.parse(ARGV)
```

と書いた方が意図が伝わりやすそうです